File: app/local_plugins/agent_translate/android/src/main/kotlin/com/aiagent/agent_translate/TranslateAgentFlutterPlugin.kt
package com.aiagent.agent_translate

import android.util.Log
import com.aiagent.plugin_interface.NativeAgentRegistry
import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * TranslateAgentFlutterPlugin — Translate Agent Flutter 插件入口
 *
 * 在 onAttachedToEngine 时注册 "translate" Agent 类型到 NativeAgentRegistry。
 * agents_server 通过 NativeAgentRegistry.create("translate") 创建 TranslateAgentSession 实例。
 */
class TranslateAgentFlutterPlugin : FlutterPlugin {

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        NativeAgentRegistry.register("translate") { TranslateAgentSession() }
        Log.d("TranslateAgentPlugin", "Registered NativeAgent type=translate")
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}

File: app/local_plugins/agents/agent_translate/ios/Classes/TranslateAgentFlutterPlugin.swift
import Flutter
import ai_plugin_interface
import os.log

/// Translate agent Flutter plugin entry.
///
/// Registers a factory for the `"translate"` agent type with the global
/// `NativeAgentRegistry`. `agents_server` instantiates a session via
/// `NativeAgentRegistry.shared.create("translate")` when the Dart side calls
/// `createAgent(agentType:"translate", …)`.
public class TranslateAgentFlutterPlugin: NSObject, FlutterPlugin {
    private static let log = OSLog(subsystem: "com.aiagent.agent_translate", category: "Plugin")

    public static func register(with registrar: FlutterPluginRegistrar) {
        NativeAgentRegistry.shared.register("translate") { TranslateAgentSession() }
        os_log("Registered NativeAgent type=translate", log: log, type: .debug)
    }
}
